dataRetriever/factory/epochProviders/currentEpochProvidersFactory_test.go

```diff
@@ -0,0 +1,57 @@
+package epochProviders
+
+import (
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/resolvers/epochproviders"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever/resolvers/epochproviders/disabled"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateCurrentEpochProvider_NilCurrentEpochProvider(t *testing.T) {
+	t.Parallel()
+
+	cnep, err := CreateCurrentEpochProvider(
+		config.Config{
+			StoragePruning: config.StoragePruningConfig{
+				FullArchive: false,
+			},
+		},
+		0,
+		0,
+	)
+
+	assert.Nil(t, err)
```

Comment:
not required, it will fail on L28 with the following error in case of nil:
```
Object expected to be of type *disabled.epochProvider, but was <nil>
```